### ISSUE TITLE
Any chance to get dynamic parameters for user procedures?

### ISSUE THREAD

**Issue by gitasmus** [dolthub/dolt#7458]
Dear Dolt team,

I would appreciate to be able to hand over parameters to procedures that can be used for SELECT options like LIMIT and OFFSET

```SQL
dolt sql -q "
 CREATE database test_procedures;
 USE test_procedures;
 CREATE table test_procedures.tab (id BIGINT NOT NULL AUTO_INCREMENT PRIMARY KEY, col INT);
 INSERT INTO test_procedures.tab(col) VALUES (1),(2),(3),(4),(5),(6);
 SELECT * FROM test_procedures.tab;
"
```
Trying to define a procedure with dynamic parameter use:
```SQL
dolt sql -q "CREATE PROCEDURE limited(the_limit INT) SELECT * FROM test_procedures.tab LIMIT the_limit;"
```
causes the error:
```
Error parsing SQL:
syntax error at position 90 near 'the_limit'
```
So, the following command cannot be executed, which would be useful for our very large tables.
```
dolt sql -q "CALL limited(3);"
```

Thank you for your help (and a great product)!

**Comment by timsehn** [comment]
We have another stored procedure request that @jycor is looking into: https://github.com/dolthub/dolt/issues/7454

We'll have him scope this one as well.

**Comment by max-hoffman** [comment]
I've linked a partial fixe for this, I still need to do error testing and figure out some edge cases around nested procedures and trigger columns. Should be able to release tomorrow.

**Comment by gitasmus** [comment]
Thank you for enabling this feature so quickly!

Solely the fact that parameters of type `unsigned bigint` raise an error, is a bit irritating as LIMIT and OFFSET should always be positive integers.

**Comment by max-hoffman** [comment]
@gitasmus Good point, fix here https://github.com/dolthub/go-mysql-server/pull/2350

**Comment by gitasmus** [comment]
Dear Maximilian,I am deeply impressed by your enthusiam, professionality and the lightspeedyness of your issue fixes!Thank you so much!AsmusAm 28.02.2024 um 17:36 schrieb Maximilian Hoffman ***@***.***>:﻿
@gitasmus Good point, fix here dolthub/go-mysql-server#2350
